File: python/test/test_custom_vector.py
import ThrustRTC as trtc

ctx = trtc.Context()

darr = trtc.device_vector(ctx, 'int32_t', 2000)
trtc.Sequence(ctx, darr)

d_custom_values = trtc.DVCustomVector(ctx, {'src': darr}, 'idx', 
'''
    unsigned group = idx / src.size();
    unsigned sub_idx = idx % src.size();
    return src[sub_idx] % (group+1) ==0 ? 1: 0;
''', 'uint32_t', darr.size()*10)

d_custom_keys = trtc.DVCustomVector(ctx, {'src': darr}, 'idx', 
'''
	return idx / src.size();
''', 'uint32_t', darr.size()*10)

d_values_out =  trtc.device_vector(ctx, 'uint32_t', 10)
d_keys_out = trtc.DVDiscard(ctx, 'uint32_t', 10)

trtc.Reduce_By_Key(ctx, d_custom_keys, d_custom_values, d_keys_out, d_values_out)

print(d_values_out.to_host())

File: java/test/test_custom_vector.java
import java.util.*;
import JThrustRTC.*;

public class test_custom_vector
{
	public static void main(String[] args) 
	{
		DVVector d_in = new DVVector(new int[] { 0, 1, 2, 3, 4 });
		DVCustomVector src = new DVCustomVector(new DeviceViewable[] { d_in }, new String[] { "src" }, "idx",
			"        return src[idx % src.size()];\n", "int32_t", d_in.size() * 5, true);
		DVVector dst= new DVVector("int32_t", 25);
		TRTC.Copy(src, dst);
		System.out.println(Arrays.toString((int[])dst.to_host()));
	}

}
